src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGImageReader.java

```diff
@@ -841,6 +845,7 @@ private void setImageData(int width,
     @Override
     public int getWidth(int imageIndex) throws IOException {
         setThreadLock();
+        verifyImageIndex(imageIndex);
```

Comment:
Dont we need to put in inside try-finally block for it to call `clearThreadLock` as it is throwing an exception?

Reply:
Yes, updated.